### PR TITLE
#184760061 - Add/View Wishlist

### PR DESCRIPTION
**Pull request by Marthatwesi**
### What does this PR do ?

A buyer should be able to add items to their Wishlist
### Description of the task completed

Given that the buyer is on a product page,
When the buyer clicks the "Add Wish" button,
Then the item should be added to the buyer's wishlist and the buyer should see a confirmation message

### How should it be tested manually 

Given that the user has logged in, the user will be able to add/remove  a product to/from their Wishlist by a button and can view it on the route /products/wishlist.

### Any background context you want to provide ?
N/A
### What are the relevant Pivotal tracker Stories

https://www.pivotaltracker.com/n/projects/2628461/stories/184760061

#### Actual
Screenshot 2023-04-25 at 17 15 05
Screenshot 2023-04-25 at 17 15 13
Screenshot 2023-04-25 at 17 15 18
Screenshot 2023-04-25 at 17 18 09
#### Figma
Screenshot 2023-04-25 at 17 21 04

### CODE REVIEW

**Review by Ntarecedrick — APPROVED** [review]

**Review by teerenzo — APPROVED** [review]

File: src/components/hooks/useSingleMainView.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+export default function UseSingleMainView() {
+    const dispatch = useDispatch();
+    const { error } = useSelector(getCart);
+    const addToCartHandler = (selectedProduct) => {
+      dispatch(addToCartThunk(selectedProduct.id)).then(() =>
+        dispatch(getCartThunk())
+      );
+      if (error.status === true) {
+        if (error.payload === 401) {
```
Comment (Eli250) [inline src/components/hooks/useSingleMainView.js:21]: Get rid of this nested if statement.

File: src/components/viewProducts/SingleMainView.js [hunk 2]
```diff
@@ -2,28 +2,28 @@ import React from "react";
...
-    if (error.status === true) {
-      if (error.payload === 401) {
-        toast.error("You need a buyer's account");
-      }
+  const {addToCartHandler, addWishlist, toggleModal, loading, added, addedWish, wishError, modal} = UseSingleMainView()
+  if (loading) {
+    return <Spinner />;
+  }
+  if (wishError.status) {
+    if (wishError.payload === 401) {
```
Comment (Eli250) [inline src/components/viewProducts/SingleMainView.js]: Remove nexted if blocks

File: src/redux/features/actions/wishlist.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+import axios from "../../axiosinstance";
+
+export const addWishListThunk = createAsyncThunk("products/addWishlist", async (id) => {
+  try {
+    const response = await axios.post("/productWishes", {
+      product_id: id,
+    },{headers:`Bearer${localStorage.getItem("token")}`});
+    return {response:response.data, productId:id};
+  } catch (error) {
+    if(error.response){
```
Comment (Eli250) [inline src/redux/features/actions/wishlist.js]: Get rid of this if block

File: src/redux/features/actions/wishlist.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+});
+/* istanbul ignore next */
+export const getSellersThunk = createAsyncThunk("cart/getSeller", async () => {
+  try {
+    const response = await axios.get("/users/sellers", {
+      headers: { Authorization: `Bearer ${localStorage.getItem("token")}` },
+    });
+    return response.data;
+  } catch (error) {
+    if (error.response) {
```
Comment (Eli250) [inline src/redux/features/actions/wishlist.js]: This if condition is redundant

File: src/redux/features/actions/wishlist.js [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+    return error.message;
+  }
+});
+export const getWishListThunk = createAsyncThunk("products/getWishlist", async () => {
+    try {
+      const response = await axios.get("/productWishes");
+      return response.data;
+    } catch (error) {
+      if(error.response){
+       return error.response.data;
```
Comment (Eli250) [inline src/redux/features/actions/wishlist.js]: Get rid of this condition

File: src/redux/features/slices/wishlist.js [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+    [addWishListThunk.pending]: (state) => {
+        state.loading = true;
+      },
+      [addWishListThunk.rejected]: (state) => {
+        state.error.status = true;
+        state.loading = false;
+      },
+      [addWishListThunk.fulfilled]: (state, { payload }) => {
+        if (payload.response.status === 201){
+            console.log("fulfilled")
```
Comment (Eli250) [inline src/redux/features/slices/wishlist.js:29]: ?

File: src/redux/features/slices/wishlist.js [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+      [getWishListThunk.pending]: (state) => {
+        state.loading = true;
+      },
+      [getWishListThunk.rejected]: (state) => {
+        state.error.status = true;
+        state.loading = false;
+      },
+      [getWishListThunk.fulfilled]: (state, { payload }) => {
+        if (payload.status === 200) {
+            console.log("fulfilled")
```
Comment (Eli250) [inline src/redux/features/slices/wishlist.js:70]: ?

**Review by Eli250 — CHANGES_REQUESTED** [review]

**Review by Eli250 — APPROVED** [review]

**Pull request merged**